### PR TITLE
check for node pools concurrently

### PR DESCRIPTION
**Pull request by Eslam-Nawara**
### Description

check for node pools concurrently to save time in case of large filtration

### Related Issues

- #497

### CODE REVIEW

File: grid-client/deployer/node_filter.go [hunk 1]
```diff
@@ -48,26 +49,46 @@ func FilterNodes(ctx context.Context, tfPlugin TFPluginClient, options types.Nod
...
+		go func(node types.Node) {
+			defer wg.Done()
+
+			client, err := tfPlugin.NcPool.GetNodeClient(tfPlugin.SubstrateConn, uint32(node.NodeID))
+			if err != nil {
+				log.Error().Err(err).Msgf("failed to get node '%d' client", node.NodeID)
+
+				lock.Lock()
+				workloads.Delete(nodePools, node)
+				lock.Unlock()
```
Comment (AbdelrahmanElawady) [inline grid-client/deployer/node_filter.go]: nodePools should only have the working nodes, we don't need the delete parts

Comment (Eslam-Nawara) [reply grid-client/deployer/node_filter.go]: I discussed it with rawda, and we didn't know why it was implemented that way with delete fa I left it as it is for now, bs it appears like there's no need for it ah

Comment (rawdaGastan) [reply grid-client/deployer/node_filter.go]: You can remove it 👍🏻

**Review by AbdelrahmanElawady — APPROVED** [review]

File: grid-client/deployer/node_filter.go [hunk 2]
```diff
@@ -48,26 +48,38 @@ func FilterNodes(ctx context.Context, tfPlugin TFPluginClient, options types.Nod
...
-	}
+	for _, node := range nodes {
+		wg.Add(1)
+
+		go func(node types.Node) {
+			defer wg.Done()
+
+			client, err := tfPlugin.NcPool.GetNodeClient(tfPlugin.SubstrateConn, uint32(node.NodeID))
+			if err != nil {
+				log.Error().Err(err).Msgf("failed to get node '%d' client", node.NodeID)
```
Comment (rawdaGastan) [inline grid-client/deployer/node_filter.go]: can we change the log level to debug (we don't want the users to get error msgs of not filtered nodes
apply for the ext condition too

**Review by rawdaGastan — APPROVED** [review]

**Pull request merged**